src/target/qemu/backlight.c

```diff
@@ -0,0 +1,75 @@
+/*
+    This project is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    Deviation is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with Deviation.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <libopencm3/stm32/gpio.h>
+#include <libopencm3/stm32/rcc.h>
+#include <libopencm3/stm32/timer.h>
+#include "common.h"
+
+// FIXME introduce constants for Timer (TIM4), compare (TIM_Oc4)
+// timer clock (RCC_APB1ENR_TIM4EN), Port (GPIOB), Pin (GPIO9)
+// Port clock RCC_APB2ENR_IOPBEN
+
+void BACKLIGHT_Init()
```

Comment:
cna u just include another real implementation?

Reply:
some of these may be able to use a real implementation, but the qemu target does not have all STM32 features, and without proper SPI or clock support a lot of things won't work.  Before I merge this, I'll clean up what can be done though.